### PR TITLE
Added missing Firework recipes

### LINKED ISSUE #48: Missing recipes for Firework Star and Firework Rocket
Firework Star has no recipe in EMI and Firework Rocket only has the simplest version of the recipe.

**List of what each of the missing recipes can contain:**

Firework Rocket:

- 1 Paper
- 1-3 Gunpowder
- 0-7 Firework Star

Firework Star:

- 1 Gunpowder
- 1-8 Dye
- 0-1 Shape effect item (head, gold nugget, feather or fire charge)
- 0-1 Diamond
- 0-1 Glowstone

Firework Star (Faded Color):

- 1 Firework Star
- 1-8 Dye

### PR DESCRIPTION
**Pull request by jopejoe1**
This adds the missing firework recipes I mentioned in #48
Closes #48

### CODE REVIEW

File: src/main/java/dev/emi/emi/recipe/special/EmiFireworkRocketRecipe.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        } else if (amount == 1) {
+            explosion.putByte("Trail", (byte) 1);
+            items++;
+        } else if (amount == 2){
+            explosion.putByte("Trail", (byte) 1);
+            explosion.putByte("Flicker", (byte) 1);
+            items = items + 2;
+        }
+
+        {
```
Comment (emilyploszaj) [inline src/main/java/dev/emi/emi/recipe/special/EmiFireworkRocketRecipe.java:129]: What's the purpose of this anon block?

File: src/main/java/dev/emi/emi/recipe/special/EmiFireworkStarFadeRecipe.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+		} else if (amount == 1) {
+			explosion.putByte("Trail", (byte) 1);
+			items++;
+		} else if (amount == 2){
+			explosion.putByte("Trail", (byte) 1);
+			explosion.putByte("Flicker", (byte) 1);
+			items = items + 2;
+		}
+
+		{
```
Comment (emilyploszaj) [inline src/main/java/dev/emi/emi/recipe/special/EmiFireworkStarFadeRecipe.java:87]: Once again curious about anon block

File: src/main/java/dev/emi/emi/recipe/special/EmiFireworkStarRecipe.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+public class EmiFireworkStarRecipe extends EmiPatternCraftingRecipe {
+    private static final List<DyeItem> DYES = Stream.of(DyeColor.values()).map(DyeItem::byColor).toList();
+
+    private static final List<Item> SHAPES = Stream.of(Items.FIRE_CHARGE, Items.FEATHER, Items.GOLD_NUGGET, Items.SKELETON_SKULL, Items.WITHER_SKELETON_SKULL, Items.CREEPER_HEAD, Items.PLAYER_HEAD, Items.DRAGON_HEAD, Items.ZOMBIE_HEAD).toList();
```
Comment (emilyploszaj) [inline src/main/java/dev/emi/emi/recipe/special/EmiFireworkStarRecipe.java:26]: Why stream just to collect to a list? List.of() is standard in the codebase.

**Review by emilyploszaj — CHANGES_REQUESTED** [review]
Overall looking really great, this is a high quality PR. I have a couple style questions and complaints I'd prefer to have resolved, but it appears otherwise ready to merge.
There were a couple instances of `.emi()` in the PR which I'd prefer to have changed to `EmiStack.of()` just for consistency as that's how it's done everywhere else.
Thank you for the PR, looking forward to merge.

File: src/main/java/dev/emi/emi/recipe/special/EmiFireworkRocketRecipe.java (hunk 1 above)
Comment (jopejoe1) [reply src/main/java/dev/emi/emi/recipe/special/EmiFireworkRocketRecipe.java]: Was an if statement, and I forgot to remove the brackets.

File: src/main/java/dev/emi/emi/recipe/special/EmiFireworkStarFadeRecipe.java (hunk 2 above)
Comment (jopejoe1) [reply src/main/java/dev/emi/emi/recipe/special/EmiFireworkStarFadeRecipe.java]: same as above

File: src/main/java/dev/emi/emi/recipe/special/EmiFireworkStarRecipe.java (hunk 3 above)
Comment (jopejoe1) [reply src/main/java/dev/emi/emi/recipe/special/EmiFireworkStarRecipe.java]: Changed it to `List.of()`

**Review by emilyploszaj — APPROVED** [review]

**Pull request merged**